### PR TITLE
blockchain: Validate stake diff in header context.

### PR DESCRIPTION
**Pull request by davecgh**
This moves the test for validating the stake difficulty specified by the header matches the calculated difficulty into the `checkBlockHeaderContext` function where it more naturally belongs since it is only dependent on the header and its contextual position within the chain.

Also, while here, it updates the error message to be more consistent with existing error messages and does not incorrectly create a rule error from an internal error in the unlikely case the difficulty fails to be calculated.

### CODE REVIEW

File: blockchain/validate.go [hunk 1]
```diff
@@ -675,11 +675,26 @@ func (b *BlockChain) checkBlockHeaderContext(header *wire.BlockHeader, prevNode
...
 			return ruleError(ErrUnexpectedDifficulty, str)
 		}
 
+		// Ensure the stake difficulty specified in the block header
+		// matches the calculated difficulty based on the previous block
+		// and difficulty retarget rules.
+		expSDiff, err := b.calcNextRequiredStakeDifficulty(prevNode)
+		if err != nil {
+			log.Errorf("calcNextRequiredStakeDifficulty: %v", err)
+			return err
```
Comment (jrick) [inline blockchain/validate.go]: why is the error both logged and returned, probably to be logged again?

Comment (davecgh) [reply blockchain/validate.go]: The error here would only be a generic failure to load a block node from the database which doesn't have any context associated with it, so the extra log here help identify the specific failure path should it be hit.

Comment (jrick) [reply blockchain/validate.go]: can more context be added to the error, instead of handling it twice?

Comment (davecgh) [reply blockchain/validate.go]: I just nuked the log message.  This is all going to change in future commits anyways such that dynamic block loading will go away.

**Pull request merged**